### ISSUE TITLE
Unexpected behavior writing booleans as 'true'

### ISSUE THREAD

**Issue by macalbert** [elastacloud/parquet-dotnet#274]
Following the issue #273, I have done a unit test that reproduces the behavior and you can prove it yourself.

This test generates a list of **CustomObjects** where they have a bool property called "**Enabled**" and are always set to "**true**", after writing this list of **CustomObjects** to file in parquet, reads the parquet data and verify that all columns are true.

Regards!

```csharp
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System.Linq;

namespace UnitTest
{
    public class CustomObject
    {
        [Newtonsoft.Json.JsonProperty("id")]
        public int Id { get; set; }
        [Newtonsoft.Json.JsonProperty("enabled")]
        public bool Enabled { get; set; }

        public CustomObject(int id, bool enabled)
        {
            this.Id = id;
            this.Enabled = enabled;
        }
    }

    public class CustomDataSet : Parquet.Data.DataSet
    {
        public CustomDataSet() : base(new Parquet.Data.SchemaElement<int>("id"), new Parquet.Data.SchemaElement<bool>("enabled"))
        {
        }

        public void Add(CustomObject co)
        {
            this.Add(co.Id, co.Enabled);
        }
    }

    [TestClass]
    public class TestClass
    {
        [TestMethod]
        public void TestMethod()
        {
            const int _numberOfItemsToGenerate = 100;
            const string _parquetFile = @"C:\data\output.parquet";

            IEnumerable<CustomObject> itemsGenerated = this.GenerateData(_numberOfItemsToGenerate);

            this.WriteFileParquet(itemsGenerated, _parquetFile);

            Parquet.Data.DataSet dataset = this.ReadParquet(_parquetFile);

            IList<int> idOfItemsAreNotTrue = new List<int>();

            for (int n = 0; n < dataset.Count; n++)
            {
                Parquet.Data.Row row = dataset.ElementAt(n);

                if (row.GetBoolean(1) != itemsGenerated.ElementAt(n).Enabled)
                {
                    idOfItemsAreNotTrue.Add(row.GetInt(0));
                }
            }

            Assert.IsTrue(idOfItemsAreNotTrue.Count == 0);
        }

        private IEnumerable<CustomObject> GenerateData(int n)
        {
            IList<CustomObject> elements = new List<CustomObject>();

            for (int i = 0; i < n; i++)
            {
                elements.Add(new CustomObject(i, true));
            }

            return elements.AsEnumerable();
        }

        private void WriteFileParquet(IEnumerable<CustomObject> list, string path)
        {
            CustomDataSet customDataSet = new CustomDataSet();
            list.ToList().ForEach(x => customDataSet.Add(x));

            using (System.IO.Stream fileStream = System.IO.File.OpenWrite(path))
            {
                using (var writer = new Parquet.ParquetWriter(fileStream))
                {
                    writer.Write(customDataSet);
                }
            }
        }

        private Parquet.Data.DataSet ReadParquet(string path)
        {
            Parquet.Data.DataSet dataset;

            using (System.IO.Stream fs = System.IO.File.OpenRead(path))
            {
                dataset = Parquet.ParquetReader.Read(fs);
            }

            return dataset;
        }
    }
}
```

**Comment by aloneguid** [comment]
Brilliant thank you, enough details to get on) this will go into 2.0 release, should be in preview shortly

**Comment by aloneguid** [comment]
Hi @macalbert I couldn't reproduce this in 2.0, I've shortened the test to:

```csharp
[Theory]
      [InlineData(100)]
      [InlineData(1000)]
      public void Write_loads_of_booleans_all_true(int count)
      {
         var ds = new DataSet(
            new DataField<int>("id"),
            new DataField<bool>("enabled"));

         //generate data
         for(int i = 0; i < count; i++)
         {
            ds.Add(i, true);
         }

         DataSet ds1 = DataSetGenerator.WriteRead(ds);

         for(int i = 0; i < count; i++)
         {
            Assert.True(ds[i].GetBoolean(1));
         }

      }
```

Can't see any difference?

**Comment by macalbert** [comment]
Hi @aloneguid great shortening of the test, but you must check **ds1** object instead of **ds**. And if you check **ds1** the test continues failing

>  Assert.True(ds[i].GetBoolean(1));

```csharp 
Assert.True(ds1[i].GetBoolean(1));
```

**Comment by aloneguid** [comment]
My bad, schoolboy error)

**Comment by aloneguid** [comment]
released hotfix as v2.0.1, hopefully this is fixed, passes all tests on my side ;)

**Comment by macalbert** [comment]
It's seems to be ok. Nice job @aloneguid!! ;)
